django_app/redbox_app/redbox_core/consumers.py

```diff
@@ -238,10 +241,24 @@ async def handle_metadata(self, response: dict):
         self.metadata = metadata_reducer(self.metadata, RequestMetadata.model_validate(response))
 
     async def handle_documents(self, response: list[Document]):
-        s3_keys = [doc.metadata["file_name"] for doc in response]
-        files = File.objects.filter(original_file__in=s3_keys)
-
+        sources = set(doc.metadata["file_name"] for doc in response)
+        files = File.objects.filter(original_file__in=sources)
```

Comment:
I wonder if it's more readable to create both the file list and the non-file-source list in a single loop, e.g.:
```python
for source in sources: 
    if File.objects.filter(original_file=source).exists:
        files.add(File.object.get(original_file=source)  # Maybe there's an all in one for this line, will have a think
    else:
        external_sources.add(doc)
```